### DISCUSSION TITLE
Why switching version from CF216 to any other versions does not retain the same page?

### DISCUSSION THREAD

**Discussion by syeda-git** [squidfunk/mkdocs-material#6809] [Q&A] [answered]
Hi,
From a CF215 page, when i switch to CF216, i get to see the same page from 216. This is good.
https://opensource.hcltechsw.com/digital-experience/CF215/build_sites/

But, from the CF216 page, when i switch to any other versions, it doesn't take me to the same page. It takes me to the Home page of that version.

As per documentation, the site_url must be set correctly. I think it is set consistently across the versions. So, not sure why only 216 has this issue.
https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#stay-on-the-same-page-when-switching-versions

Any idea what's going on?

Syed

**Comment by kamilkrzyskow** [comment]
It's up to you to debug it and provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) and since it requires mike you should also create a minimal reproduction repository with deployed GitHub Pages.
You didn't do that in any of the previously reported issues:
- https://github.com/squidfunk/mkdocs-material/issues/6657
- https://github.com/squidfunk/mkdocs-material/issues/6716

Looking at the source of the page, I see that CF216 was created using the latest 9.5.10 version, and all other versions, even CF217 uses 8.5.10. So it's likely related to the changes in instant navigation, similar to:
- https://github.com/squidfunk/mkdocs-material/discussions/6788

In this discussion:
- https://github.com/squidfunk/mkdocs-material/discussions/6537

I mentioned there was some kind of instant loading version switcher interception logic:
In both 8.5.10 and 9.5.10 this code is still available:
https://github.com/squidfunk/mkdocs-material/blob/8.5.10/src/assets/javascripts/integrations/version/index.ts#L93-L146
https://github.com/squidfunk/mkdocs-material/blob/9.5.10/src/templates/assets/javascripts/integrations/version/index.ts#L93-L146

**Reply by kamilkrzyskow** [reply]
I thought a bit what could be the issue, and the idea crossed my mind, that the sitemap could be incorrect.
https://opensource.hcltechsw.com/digital-experience/CF216/sitemap.xml uses the `/latest/` version identifier, same for CF215, so the idea seemed wrong, but I double checked the differences between 8.5.10 and 9.5.10.

The code I linked above, actually has a single different line:
https://github.com/squidfunk/mkdocs-material/blob/08bf99265e51abdb3f86fd4f573bd76caa25fb9e/src/assets/javascripts/integrations/version/index.ts#L137
https://github.com/squidfunk/mkdocs-material/blob/b688cfdd09bab236f2b0126f94d116b98d5240ce/src/templates/assets/javascripts/integrations/version/index.ts#L137

And this lead me to find that sitemap logic got changed between those versions:
https://github.com/squidfunk/mkdocs-material/blob/8.5.10/src/assets/javascripts/integrations/sitemap/index.ts
https://github.com/squidfunk/mkdocs-material/blob/9.5.10/src/templates/assets/javascripts/integrations/sitemap/index.ts

Perhaps now, because the sitemap includes the `/latest/` links the /CF216/ created with 9.5.10 doesn't see /CF215/ as an internal URL 🤔 
I didn't debug further and I'm not planning to, but I consider it a fair idea worth mentioning.

**Reply by squidfunk** [reply]
Thanks for investigating. I've refactored the sitemap logic as part of the latest instant navigation refactoring (fixing _all_ the bugs, hopefully!) and [now uses a `Map`](https://github.com/squidfunk/mkdocs-material/blob/59f4147e6026ca02d08de2935d9036947e1e4b68/src/templates/assets/javascripts/integrations/sitemap/index.ts#L43) instead of an array in order to add support for switching between alternate versions (in Insiders). That's why the logic changed, but other than that, no change to versioning happened.

We definitely need a minimal reproduction to go forward, as mentioned several times.

**Comment by squidfunk** [comment]
@syeda-git In fact, @kamilkrzyskow is right. You are binding precious time of our contributors and community experts without actually listening to them, expecting that we help you for free without investing at least some effort. We are only as good as the information we are given, and we don't have time to debug the repositories or even proprietary pages without source being available.

[This is your first warning](https://squidfunk.github.io/mkdocs-material/contributing/#fn:1).

I kindly ask you to study our mandatory [bug reporting guide](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/) and follow the exact steps when you report bugs. Discussions are not for bug reports. You can ask questions here, but in order to determine if some behavior is a bug, let alone fix it, a complete bug report is required. Please respect that. After all, we are helping you _for free_. Thank you.

**Reply by syeda-git** [reply]
@squidfunk I understand bug reporting has certain prerequisites. Before relating a behavior to a bug, this discussion was meant to ask a question about the behavior. Has anything changed by designed in the latest version. I am not sure if opening a discussion too has certain prerequisites. If there are, it would be good to show that to the users when they open a discussion. Or, if the preference is to first open a bug, please make it clear. Let me know if i am missing something.

Your offering and service _is free_. I value that. Please note that I as an employee in a company can only tell my management to sponsor the project. My responsibility ends there.

**Reply by squidfunk** [reply, marked as answer]
You've created two previous issues with essentially the same question, as @kamilkrzyskow pointed out, and did not follow up the requests for minimal reproductions. One of your co-workers [created another duplicate discussion](https://github.com/squidfunk/mkdocs-material/discussions/6810) 2 hours after you created this discussion, all asking essentially the same question. We are losing precious time answering those incomplete reports – time we could otherwise invest into helping other users, actually _fixing_ bugs, or implementing new useful features.

> Or, if the preference is to first open a bug, please make it clear. Let me know if i am missing something.

I'm not sure what is unclear here. When you open an issue you agree (checkbox!) that you've carefully read our [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/), which you obviously did not, because I repeatedly had to ask for reproductions which you did not provide. Minimal reproductions are mandatory. So, again, please carefully study our bug reporting guide and we are happy to help you.

> Your offering and service is free. I value that. Please note that I as an employee in a company can only tell my management to sponsor the project. My responsibility ends there.

You have the responsibility to provide us with complete information, so we can actually help you. You did not do that repeatedly, which is why I issued the warning. We are always happy to help our users, if they make it possible for us to do.

**Reply by squidfunk** [reply]
I'm locking this discussion, as all has been said.
